File: bintree_py/test/testavl02.py
from .testbase import print_tree
from binarytree import AvlTree


def do_task():
    avl = AvlTree(lst=[10, 20, 30, 40, 50, 25])

    print('\n size:', avl.size())
    print('\n min:', avl.min())
    print('\n max:', avl.max())
    print('\n contain:', avl.contain(50))
    print('\n height:', avl.height())

    print('\n print tree')
    print_tree(avl)

    print()

File: bintree_csharp/test/TestAvl02.cs
using System;
using System.Collections.Generic;
using my.binarytree;


namespace my.test
{
    class TestAvl02 : TestBase
    {
        public override void doTask()
        {
            var avl = new AvlTree<int>(new List<int>(new int[] { 10, 20, 30, 40, 50, 25 }));

            Console.WriteLine("\n count: " + avl.Count);
            Console.WriteLine("\n min: " + avl.Min());
            Console.WriteLine("\n max: " + avl.Max());
            Console.WriteLine("\n contain: " + avl.Contain(50));
            Console.WriteLine("\n height: " + avl.Height);

            Console.WriteLine("\n print tree: ");
            PrintTree(avl);

            Console.WriteLine();
        }
    }
}
